src/app/common/services/pdf.service.ts

```diff
@@ -17,39 +17,71 @@ export class PdfService {
 		protected http: HttpClient,
 		protected configService: AppConfigService,
 		protected messageService: MessageService,
-    private sanitizer: DomSanitizer
+		private sanitizer: DomSanitizer,
 	) {
-    this.baseUrl = this.configService.apiConfig.baseUrl
-  }
+		this.baseUrl = this.configService.apiConfig.baseUrl;
+	}
 
 	getPdf(slug: string): Promise<SafeResourceUrl> {
-        const headers = new HttpHeaders().set('Authorization', `Bearer ${this.loginService.currentAuthToken.access_token}`);
-        return this.http.get(`${this.baseUrl}/v1/earner/badges/pdf/${slug}`, { headers: headers, responseType: 'blob' }).pipe(
-          map((response: Blob) => {
-            const url = URL.createObjectURL(response);
-            // sanitize the url before avoiding security check
-            const safe_url = this.sanitizer.sanitize(SecurityContext.URL, url);
-            return this.sanitizer.bypassSecurityTrustResourceUrl(safe_url);
-          })
-        ).toPromise();
-  }
-  
-  downloadPdf(pdfSrc: SafeResourceUrl, badgeName: string, issueDate: Date) {
-    const link = document.createElement('a');
-    // https://stackoverflow.com/questions/55849415/type-saferesourceurl-is-not-assignable-to-type-string
-    const url = this.sanitizer.sanitize(SecurityContext.RESOURCE_URL, pdfSrc);
-    link.href = url;
-    link.download = badgeName + ' - ' + this.dateToString(issueDate, '') + '.pdf';
-    link.click();
-  }
+		const headers = new HttpHeaders().set(
+			'Authorization',
+			`Bearer ${this.loginService.currentAuthToken.access_token}`,
+		);
+		return this.http
+			.get(`${this.baseUrl}/v1/earner/badges/pdf/${slug}`, { headers: headers, responseType: 'blob' })
+			.pipe(
+				map((response: Blob) => {
+					const url = URL.createObjectURL(response);
+					// sanitize the url before avoiding security check
+					const safe_url = this.sanitizer.sanitize(SecurityContext.URL, url);
+					return this.sanitizer.bypassSecurityTrustResourceUrl(safe_url);
+				}),
+			)
+			.toPromise();
+	}
+	// new function getCertificatepdf which takes the slug and a base64 image string as input
+	// does a post request and receives a pdf blob as response
+	getCertificatePdf(slug: string, base64Image: string): Promise<SafeResourceUrl> {
```

Comment:
Would it be possible to only have merge both ```getPdf```functions into only one?

Reply:
Yes. I edited `src/app/common/dialogs/export-pdf-dialog/export-pdf-dialog.component.ts`accordingly, however, I was not able to find out where exactly this dialog option is used in the ui to test it